src/borg/archiver.py

```diff
@@ -1265,6 +1268,7 @@ def compare_or_defer(item1, item2):
             else:
                 print_output = print_text_output
 
+            # if sorted, then results are saved in output, otherwise output is empty and results already printed.
```

Comment:
somehow that comment does not fit the code.

sorted() is not checking whether it is sorted (but does return sorted result), so "if sorted ..." sounds strange.

Reply:
maybe rather:

if we want sorted output (args.sort is true), we have collected stuff in output, otherwise results were already printed and output is empty.

Reply:
ok, updated and also pushed some tests and docs. Haven't looked thoroughly yet to see if I've caught all the places that need documenting. And I need to figure out how to do a complete build of everything - python is not my daily language and normally just small single file type scripting, so still trying to figure some things out here.

Reply:
there is a section for developers in the docs. the basic installation steps are same as in the install section.

I'ld recommend using a virtual env and `pip install -e .`.